spring-boot-devtools do not work together with declarative files

If you have a Spring Boot project and include `spring-boot-devtools`, Spring will use a `RestartClassLoader´ to load application classes so that it can throw away that class loader and all application classes when some part of the application changes. If you use a declarative design file and include a (custom) component from the project itself, you will get an error while binding that component to a field:
```
Caused by: java.lang.IllegalArgumentException: Can not set my.project.MyCustomComponent field my.project.DesignFile.customComponentField to my.project.MyCustomComponent
```

The field `type` is a class loaded by `RestartClassLoader` but the instance created by `Design.read` is created by the parent class loader:
```
instance.getClass().getClassLoader()
	 (sun.misc.Launcher$AppClassLoader) sun.misc.Launcher$AppClassLoader@18b4aac2
field.getType().getClassLoader()
	 (org.springframework.boot.devtools.restart.classloader.RestartClassLoader) org.springframework.boot.devtools.restart.classloader.RestartClassLoader@7662323c
```

--- a/server/src/main/java/com/vaadin/server/VaadinServiceClassLoaderUtil.java
+++ b/server/src/main/java/com/vaadin/server/VaadinServiceClassLoaderUtil.java
@@ -24,7 +24,7 @@ import java.security.PrivilegedAction;
  * @since 7.4
  * @author Vaadin Ltd
  */
-class VaadinServiceClassLoaderUtil {
+public class VaadinServiceClassLoaderUtil {
 
     private static class GetClassLoaderPrivilegedAction
             implements PrivilegedAction<ClassLoader> {
@@ -48,7 +48,7 @@ class VaadinServiceClassLoaderUtil {
      *             if current security policy doesn't allow acquiring current
      *             thread's context class loader
      */
-    static protected ClassLoader findDefaultClassLoader()
+    public static ClassLoader findDefaultClassLoader()
             throws SecurityException {
         return AccessController.doPrivileged(
                 new VaadinServiceClassLoaderUtil.GetClassLoaderPrivilegedAction());

--- a/server/src/main/java/com/vaadin/ui/declarative/Design.java
+++ b/server/src/main/java/com/vaadin/ui/declarative/Design.java
@@ -36,6 +36,7 @@ import org.jsoup.parser.Parser;
 import org.jsoup.select.Elements;
 
 import com.vaadin.annotations.DesignRoot;
+import com.vaadin.server.VaadinServiceClassLoaderUtil;
 import com.vaadin.shared.util.SharedUtil;
 import com.vaadin.ui.Component;
 import com.vaadin.ui.declarative.DesignContext.ComponentCreatedEvent;
@@ -192,7 +193,9 @@ public class Design implements Serializable {
         protected Class<? extends Component> resolveComponentClass(
                 String qualifiedClassName, DesignContext context) {
             try {
-                Class<?> componentClass = Class.forName(qualifiedClassName);
+                Class<?> componentClass = Class.forName(qualifiedClassName,
+                        true,
+                        VaadinServiceClassLoaderUtil.findDefaultClassLoader());
                 return componentClass.asSubclass(Component.class);
             } catch (ClassNotFoundException e) {
                 throw new DesignException("Unable to load component for design",

